assert/assertions.go

```diff
@@ -1884,11 +1890,20 @@ func (c *CollectT) Copy(t TestingT) {
 	if tt, ok := t.(tHelper); ok {
 		tt.Helper()
 	}
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 	for _, err := range c.errors {
 		t.Errorf("%v", err)
 	}
 }
 
+// HasErrors returns true if any errors were collected.
+func (c *CollectT) HasErrors() bool {
```

Comment:
`HasErrors` should be private (`Reset` and `Copy` should have been made private too unfortunately, but they are now part of the public API).